adapters/sidekiq/lib/opentelemetry/adapters/sidekiq/adapter.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Sidekiq
+      # The Adapter class contains logic to detect and install the Sidekiq
+      # instrumentation adapter
+      class Adapter < OpenTelemetry::Instrumentation::Adapter
+        install do |_config|
+          require_dependencies
+          add_client_middleware
+          add_server_middleware
+        end
+
+        present do
+          defined?(::Sidekiq)
+        end
+
+        private
+
+        def require_dependencies
+          require_relative 'middlewares/client/tracer_middleware'
+          require_relative 'middlewares/server/tracer_middleware'
+        end
+
+        def add_client_middleware
+          ::Sidekiq.configure_client do |config|
+            config.client_middleware do |chain|
+              chain.add Middlewares::Client::TracerMiddleware
+            end
+          end
+        end
+
+        def add_server_middleware
+          ::Sidekiq.configure_server do |config|
+            config.server_middleware do |chain|
+              chain.add Middlewares::Server::TracerMiddleware
+            end
+          end
+
+          ::Sidekiq.configure_client do |config|
```

Comment:
It looks like we add Client::TracerMiddleware twice. Once here and once on line 31. Is that intentional?

Reply:
Adding it twice is intentional. See https://github.com/mperham/sidekiq/wiki/Middleware#client-middleware-registered-in-both-places. It is done incorrectly here, though. The `config.client_middleware` below should be in the `configure_server` block above.

Reply:
Thanks for catching that, it looks like `Sidekiq::Testing.inline!` was hiding the problem a bit here as it was still producing the spans in the way I would expect them to (for the most part).  

I've since fixed this (and the tests) as well as tested it against a simple rails app to confirm it's behaving the way it should be.